Converted to Guzzle 6 and added Shopify example

CDyWeb opened the pull request.

fkooman commented:
Hi, 

Thanks for your pull request! There is a slight problem merging this, as Travis-CI doesn't seem to work anymore, possibly due to the PHP 5.3 test it runs :)

So, I'd be okay with supporting PHP 5.4 as a minimum, but 5.5 is too "new" for some of my deploys as CentOS 7 (RHEL 7) has PHP 5.4 only (php-5.4.16-23.el7_0.3.x86_64).

So an upgrade to Guzzle 4 would be okay, but it seems Guzzle 6 requires at least PHP 5.5 which is a no-go for me. 

So ideally we'd refactor everything so there is a way to abstract the version of Guzzle in a separate class with a common interface so we can support both Guzzle 3 and Guzzle 6 using DI or something... That looks like an interesting approach. 

Also, it would be great if your pull request would do one thing at a time, so adding Shopify is separate from upgrading Guzzle... 

What do you think?

fkooman commented:
Hi, 

Thanks for your pull request! There is a slight problem merging this, as Travis-CI doesn't seem to work anymore, possibly due to the PHP 5.3 test it runs :)

So, I'd be okay with supporting PHP 5.4 as a minimum, but 5.5 is too "new" for some of my deploys as CentOS 7 (RHEL 7) has PHP 5.4 only (php-5.4.16-23.el7_0.3.x86_64).

So an upgrade to Guzzle 4 would be okay, but it seems Guzzle 6 requires at least PHP 5.5 which is a no-go for me. 

So ideally we'd refactor everything so there is a way to abstract the version of Guzzle in a separate class with a common interface so we can support both Guzzle 3 and Guzzle 6 using DI or something... That looks like an interesting approach. 

Also, it would be great if your pull request would do one thing at a time, so adding Shopify is separate from upgrading Guzzle... 

What do you think?

CDyWeb commented:
Sounds good, I'll see what I can do.

Erwin

On 06/06/2015 7:00 AM, François Kooman wrote:
>
> Hi,
>
> Thanks for your pull request! There is a slight problem merging this, 
> as Travis-CI doesn't seem to work anymore, possibly due to the PHP 5.3 
> test it runs :)
>
> So, I'd be okay with supporting PHP 5.4 as a minimum, but 5.5 is too 
> "new" for some of my deploys as CentOS 7 (RHEL 7) has PHP 5.4 only 
> (php-5.4.16-23.el7_0.3.x86_64).
>
> So an upgrade to Guzzle 4 would be okay, but it seems Guzzle 6 
> requires at least PHP 5.5 which is a no-go for me.
>
> So ideally we'd refactor everything so there is a way to abstract the 
> version of Guzzle in a separate class with a common interface so we 
> can support both Guzzle 3 and Guzzle 6 using DI or something... That 
> looks like an interesting approach.
>
> Also, it would be great if your pull request would do one thing at a 
> time, so adding Shopify is separate from upgrading Guzzle...
>
> What do you think?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/fkooman/php-oauth-client/pull/49#issuecomment-109564772>.
>

fkooman commented:
Hi,

I've been working on extracting Guzzle to a separate interface so it becomes easy to allow both Guzzle 3 and Guzzle 6. I've implemented the Guzzle3Client now.

https://github.com/fkooman/php-oauth-client/blob/guzzle-wrapper/src/fkooman/OAuth/Client/Guzzle3Client.php

This class implements this interface:

https://github.com/fkooman/php-oauth-client/blob/guzzle-wrapper/src/fkooman/OAuth/Client/HttpClientInterface.php

I've updated the unit tests, but not yet the examples to use this. You could now easily implement a Guzzle6Client that you feed to the `Api` and `Callback` classes from your application. 

The unit tests use Guzzle3Client as that seemed the easiest as they use the logging/history plugins for Guzzle3.

CDyWeb commented:
Perfect! I will have a look later this week.
Thanks
Erwin

On 10/06/2015 4:42 AM, François Kooman wrote:
>
> Hi,
>
> I've been working on extracting Guzzle to a separate interface so it 
> becomes easy to allow both Guzzle 3 and Guzzle 6. I've implemented the 
> Guzzle3Client now.
>
> https://github.com/fkooman/php-oauth-client/blob/guzzle-wrapper/src/fkooman/OAuth/Client/Guzzle3Client.php
>
> This class implements this interface:
>
> https://github.com/fkooman/php-oauth-client/blob/guzzle-wrapper/src/fkooman/OAuth/Client/HttpClientInterface.php
>
> I've updated the unit tests, but not yet the examples to use this. You 
> could now easily implement a Guzzle6Client that you feed to the |Api| 
> and |Callback| classes from your application.
>
> The unit tests use Guzzle3Client as that seemed the easiest as they 
> use the logging/history plugins for Guzzle3.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/fkooman/php-oauth-client/pull/49#issuecomment-110656762>.
>

CDyWeb commented:
I have refactored the HttpClientInterface structure to use an external, reusable lib, and updated the examples and unit tests. The oauth lib is now completely Guzzle version independent, it can work with version 3,4,5, and 6. Please have a look. I don't understand why the CI tests fail, IMO it is all PHP5.3 compatible now.

fkooman commented:
Thanks for your effort! I am not making it easy :(

I took your ShopifyClientConfig class, fixed my simple example and added the simple6 sample and merged it back to master. Still have to fix the gdrive example. 

I like the idea of making it completely guzzle version independent, but it would only be relevant for the tests... which currently work fine with Guzzle3.

As for the other things, it would be cool to have one pull request per issue/change and discuss the usefulness of it over there, for example the session close thing. It is a bit hard to pull from this and figure out the need for everything...
